assets/js/modules/analytics-4/components/widgets/TopDeviceDrivingPurchasesWidget.js

```diff
@@ -0,0 +1,217 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect, useInViewSelect } from 'googlesitekit-data';
+import MetricTileText from '../../../../components/KeyMetrics/MetricTileText';
+import {
+	CORE_USER,
+	KM_ANALYTICS_TOP_DEVICE_DRIVING_PURCHASES,
+} from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
+import { numFmt } from '../../../../util';
+import { get } from 'lodash';
+import whenActive from '../../../../util/when-active';
+import ConnectGA4CTATileWidget from './ConnectGA4CTATileWidget';
+
+function TopDeviceDrivingPurchases( { Widget } ) {
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+			compare: true,
+		} )
+	);
+
+	const detectedEvents = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getDetectedEvents()
+	);
+	const hasDetectedEvent = detectedEvents?.includes( 'purchase' );
+
+	const totalPurchasesReportOptions = {
+		...dates,
+		metrics: [
+			{
+				name: 'ecommercePurchases',
+			},
+		],
+	};
+
+	const deviceReportOptions = {
+		...dates,
+		dimensions: [ 'deviceCategory' ],
+		metrics: [
+			{
+				name: 'ecommercePurchases',
+			},
+		],
+		limit: 1,
+		orderBy: 'ecommercePurchases',
+	};
+
+	const totalPurchasesReport = useInViewSelect(
+		( select ) =>
+			hasDetectedEvent
+				? select( MODULES_ANALYTICS_4 ).getReport(
+						totalPurchasesReportOptions
+				  )
+				: undefined,
+		[ totalPurchasesReportOptions ]
+	);
+
+	// ecommercePurchases metric will always be assigned a value, unlike most
+	// other metrics, where rows would be empty if there is no data, ecommercePurchases
+	// will assign value `0` for each date range if there is no event data. So we need
+	// to verify that value is non-zero for one of the rows, to allow report request for
+	// traffic source. Otherwise zero data will be incorrectly displayed, showing zero for
+	// purchases and comparison, but still include the traffic source.
+	const hasPurchases = totalPurchasesReport?.rows
+		? totalPurchasesReport?.rows?.[ 0 ]?.metricValues?.[ 0 ]?.value > 0 ||
+		  totalPurchasesReport?.rows?.[ 1 ]?.metricValues?.[ 0 ]?.value > 0
+		: false;
+
+	const deviceReport = useInViewSelect(
+		( select ) =>
+			hasDetectedEvent && hasPurchases
+				? select( MODULES_ANALYTICS_4 ).getReport( deviceReportOptions )
+				: undefined,
+		[ deviceReportOptions ]
```

Comment:
`hasDetectedEvent` and `hasPurchases` are also missing in the dependencies array.

I will commit these changes, since they are tiny to avoid another iteration for this.